### PR TITLE
Use explicit LIMIT in scheduler task polling query for PostgreSQL

### PR DESCRIPTION
**Pull request by rafal-kowalski**
Fixes behaviour described in https://github.com/kagkarlsson/db-scheduler/issues/164

### CODE REVIEW

**Comment by rafal-kowalski** [comment]
Ready for code review

File: db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java [hunk 1]
```diff
@@ -135,9 +135,9 @@ public void getScheduledExecutions(ScheduledExecutionsFilter filter, String task
     @Override
     public List<Execution> getDue(Instant now, int limit) {
         final UnresolvedFilter unresolvedFilter = new UnresolvedFilter(taskResolver.getUnresolved());
-
+        final String explicitLimit = jdbcCustomization.supportsExplicitQueryLimitPart() ? jdbcCustomization.getQueryLimitPart(limit) : "";
         return jdbcRunner.query(
-                "select * from " + tableName + " where picked = ? and execution_time <= ? " + unresolvedFilter.andCondition() + " order by execution_time asc",
+                "select * from " + tableName + " where picked = ? and execution_time <= ? " + unresolvedFilter.andCondition() + " order by execution_time asc" + explicitLimit,
```
Comment (kagkarlsson) [inline db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java:140]: I would be tempted to drop `setMaxRows(...)` when explicit limit in sql is used

**Review by kagkarlsson — COMMENTED** [review]
Looks good 👍  
Just one small comment

File: db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java (hunk 1 above)
Comment (rafal-kowalski) [reply db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java:140]: Ok, I've removed `setMaxRows` when there is explicit limit

**Review by kagkarlsson — APPROVED** [review]
Looks good 👍

**Comment by kagkarlsson** [comment]
Thank you for contributing!

**Pull request merged**